src/main/java/com/exasol/adapter/dynamodb/mapping/AbstractTableMappingFactory.java

```diff
@@ -0,0 +1,104 @@
+            tables.addAll(nestedTableReader.readNestedTable(rootTable));
+        }
+        return tables;
+    }
+
+    protected abstract TableMappingDefinition readTable(final JsonObject definition);
+
+    protected final void visitMapping(final JsonObject definition, final DocumentPathExpression.Builder sourcePath,
+            final TableMappingDefinition.Builder tableBuilder, final String propertyName, final boolean isRootLevel) {
+        final JsonColumnMappingFactory columnMappingFactory = new JsonColumnMappingFactory();
+        switch (getMappingType(definition)) {
+        case TO_STRING_MAPPING_KEY:
+            columnMappingFactory.addStringColumnIfPossible(definition.getJsonObject(TO_STRING_MAPPING_KEY), sourcePath,
+                    tableBuilder, propertyName, isRootLevel);
+            break;
+        case TO_JSON_MAPPING_KEY:
+            columnMappingFactory.addToJsonColumn(definition.getJsonObject(TO_JSON_MAPPING_KEY), sourcePath,
+                    tableBuilder, propertyName);
+            break;
+        case TO_TABLE_MAPPING_KEY:
+            queueAddingNestedTable(definition.getJsonObject(TO_TABLE_MAPPING_KEY), sourcePath, propertyName);
+            break;
+        case FIELDS_KEY:
+            visitChildren(definition.getJsonObject(FIELDS_KEY), sourcePath, tableBuilder);
+            break;
+        case "":// no mapping definition
+            break;
+        default:
+            throw new UnsupportedOperationException("This mapping type is not supported in the current version.");
+        }
+    }
+
+    private void queueAddingNestedTable(final JsonObject definition, final DocumentPathExpression.Builder sourcePath,
+            final String propertyName) {
+        this.nestedTableReaderQueue
+                .add(parentTable -> new NestedTableMappingFactory(parentTable, propertyName, sourcePath)
+                        .readMappingDefinition(definition));
+    }
+
+    private String getMappingType(final JsonObject definition) {
+        final Set<String> keys = definition.keySet();
+        if (keys.isEmpty()) {
+            return "";
+        } else if (keys.size() == 1) {
+            return keys.iterator().next();
+        } else {
+            throw new ExasolDocumentMappingLanguageException("Please, define only one mapping for one property.");
+        }
+    }
+
+    private void visitChildren(final JsonObject definition, final DocumentPathExpression.Builder sourcePath,
+            final TableMappingDefinition.Builder tableBuilder) {
+        for (final String dynamodbPropertyName : definition.keySet()) {
+            final DocumentPathExpression.Builder newBuilder = new DocumentPathExpression.Builder(sourcePath)
+                    .addObjectLookup(dynamodbPropertyName);
+            visitMapping(definition.getJsonObject(dynamodbPropertyName), newBuilder, tableBuilder, dynamodbPropertyName,
+                    false);
+        }
+    }
+
+    protected static interface NestedTableReader {
```

Comment:
No class implements this interface yet, but I guess that is expected at the moment?

Reply:
No, it is a used by the Labda Expression in line 76. To clarify that it is a [functional interface](https://www.baeldung.com/java-8-lambda-expressions-tips) I will anotate it with @FunctionalInterface. Even though Baeldug suggests to use Fuctiont<T,R> if possible I prefered this custom interface as it adds more meaning, if the method has a good name. Even more i did just made the interface private to signal it is just used in this class.

Reply:
This is very new to me. Thanks for the clarification!